aten/src/ATen/cpu/vec256/vec256_base.h

```diff
@@ -64,115 +66,199 @@ struct Vec256 {
-  }
-  Vec256<T> cos() const {
-    return map(std::cos);
-  }
-  Vec256<T> floor() const {
-    return map(std::floor);
-  }
-  Vec256<T> round() const {
-    return map(std::round);
-  }
-  Vec256<T> sin() const {
-    return map(std::sin);
+};
+template <class T>Vec256<T> map(T (*f)(T), Vec256<T> x) {
+  Vec256<T> ret;
+  for (int64_t i = 0; i != Vec256<T>::size; i++) {
+    ret.set_value(i, f(x[i]));
   }
-  Vec256<T> tanh() const {
-    return map(std::tanh);
+  return ret;
+}
+
+template <class T>Vec256<T> abs(Vec256<T> x) {
+  Vec256<T> ret;
+  for (int64_t i = 0; i < Vec256<T>::size; i++) {
+    ret.set_value(i, x[i] < 0 ? -x[i] : x[i]);
   }
-  Vec256<T> trunc() const {
-    return map(std::trunc);
+  return ret;
+}
+
+template <class T>
+Vec256<T> acos(Vec256<T> x) {
+  return map(std::acos, x);
+}
+
+template <class T>
+Vec256<T> asin(Vec256<T> x) {
+  return map(std::asin, x);
+}
+
+template <class T>
+Vec256<T> atan(Vec256<T> x) {
+  return map(std::atan, x);
+}
+
+template <class T>
+Vec256<T> erf(Vec256<T> x) {
+  return map(std::erf, x);
+}
+
+template <class T>
+Vec256<T> exp(Vec256<T> x) {
+  return map(std::exp, x);
+}
+
+template <class T>
+Vec256<T> expm1(Vec256<T> x) {
+  return map(std::expm1, x);
+}
+
+template <class T>
+Vec256<T> log(Vec256<T> x) {
+  return map(std::log, x);
+}
+
+template <class T>
+Vec256<T> log10(Vec256<T> x) {
+  return map(std::log10, x);
+}
+
+template <class T>
+Vec256<T> log1p(Vec256<T> x) {
+  return map(std::log1p, x);
+}
+
+template <class T>
+Vec256<T> log2(Vec256<T> x) {
+  return map(std::log2, x);
+}
+
+template <class T>
+Vec256<T> ceil(Vec256<T> x) {
+  return map(std::ceil, x);
+}
+
+template <class T>
+Vec256<T> cos(Vec256<T> x) {
+  return map(std::cos, x);
+}
+
+template <class T>
+Vec256<T> floor(Vec256<T> x) {
+  return map(std::floor, x);
+}
+
+template <class T>
+Vec256<T> round(Vec256<T> x) {
+  return map(std::round, x);
+}
+
+template <class T>
+Vec256<T> sin(Vec256<T> x) {
+  return map(std::sin, x);
+}
+
+template <class T>
+Vec256<T> sqrt(Vec256<T> x) {
+  return map(std::sqrt, x);
+}
+
+template <class T>
+Vec256<T> neg(Vec256<T> x) {
+  return Vec256<T>(0) - x;
+}
+
+template <class T>
+Vec256<T> reciprocal(Vec256<T> x) {
+  Vec256<T> ret;
+  for (int64_t i = 0; i < Vec256<T>::size; i++) {
+    ret.set_value(i, 1 / x[i]);
   }
-  Vec256<T> sqrt() const {
-    return map(std::sqrt);
+  return ret;
+}
+
+template <class T>
+Vec256<T> rsqrt(Vec256<T> x) {
+  return reciprocal(sqrt(x));
+}
+
+template <class T>
+Vec256<T> sigmoid(Vec256<T> x) {
+  Vec256<T> ret;
+  for (int64_t i = 0; i < Vec256<T>::size; i++) {
+    ret.set_value(i, ((T)1.0) / (((T)1.0) + std::exp(-x[i])));
   }
-};
+  return ret;
+}
+
+template <class T>
+Vec256<T> tanh(Vec256<T> x) {
+  return map(std::tanh, x);
+}
+
+template <class T>
+Vec256<T> trunc(Vec256<T> x) {
+  return map(std::trunc, x);
+}
+
+template <class T>
+Vec256<T> frac(Vec256<T> x) {
+  return x - trunc(x);
+}
```

Comment:
Are any of those used in the vectorized impls?

Reply:
Yes, this is.